src/js/app.js

```diff
@@ -1,6 +1,710 @@
+  
+    // Отримайте tbody для таблиці
+    const tbody = table.querySelector("tbody");
+  
+    // Функція для оновлення таблиці
+function updateTable(sortOrder, order) {
+       
+    while (tbody.firstChild) {
+            tbody.removeChild(tbody.firstChild);
+          }
+           // Початковий порядок сортування (за ім'ям, спеціальністю, країною та віком)
+          
+        let sortedTeacher = finalArray;
+        sortedTeacher = sortObjects(finalArray, sortOrder, order);
+      
+      // Додайте відсортовані дані до таблиці
+      sortedTeacher.forEach((item) => { 
+        const row = document.createElement("tr");
+        const keys = ["full_name", "course", "age", "gender", "country"];
+        keys.forEach((key) => {
+          const cell = document.createElement("td");
+          cell.textContent = item[key];
+          row.appendChild(cell);
+        });
+        tbody.appendChild(row);
+      });
+     
+    }
+    let order = 1;
+    // Функція для зміни sortOrder при кліку на заголовок стовпця
+    function handleHeaderClick(event) {
+      const clickedHeader = event.target.closest("th");
+      if (!clickedHeader) return;
+  
+      // Отримайте текс з заголовка
+      const headerText = clickedHeader.textContent.trim();
+      let sortOrder = "";
+     
+      // Визначте sortOrder на основі тексту заголовка
+      if (headerText === "Name ↓") sortOrder = "full_name";
+      else if (headerText === "Speciality ↓") sortOrder = "course";
+      else if (headerText === "Age ↓") sortOrder = "age";
+      else if (headerText === "Nationality ↓") sortOrder = "country";
+  
+
+      // Оновіть таблицю
+      updateTable(sortOrder, order*=-1);
+    }
+  
+    // Додайте обробник подій для заголовків стовпців
+    headers.forEach((header) => {
+      header.addEventListener("click", handleHeaderClick);
+    });
+  
+    // Початкове заповнення та сортування таблиці
+    updateTable(0);
+}
+
+function handleAddTeacherButtons(){
+    const addTeacherButtons = document.querySelectorAll('.addTeacherButton');
+    addTeacherButtons.forEach((button) => {button.addEventListener("click", () => {
+        showAddTeacherPopup();
+         });
+    });
+    const closePopupAddTeacher =  document.getElementById('closePopupAddTeacher');
+    closePopupAddTeacher.addEventListener("click",() => {
+        const teacherPopup = document.getElementById('addTeacherPopupBack');
+        teacherPopup.style.display="none";
+    } );
+}
+function showAddTeacherPopup(){
+    const teacherPopup = document.getElementById('addTeacherPopupBack');
+    teacherPopup.style.display="block";
+}
+
+
+
+function searchTeacherByValue(arr, value) {
+    return arr.filter((obj) => { 
+       if (typeof value === 'string') {
+            if((obj["full_name"]).toLowerCase().includes(value.toLowerCase()) || (obj["note"]).toLowerCase().includes(value.toLowerCase())){
+                return true;
+            }else return String(obj["age"]).includes(String(value));
```

Comment:
For better readability - use brackets everywhere
```suggestion
            }else { return String(obj["age"]).includes(String(value)) };
```